dot/network/messages/warp_sync.go

```diff
@@ -0,0 +1,51 @@
+// Copyright 2024 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package messages
+
+import (
+	"bytes"
+	"fmt"
+
+	"github.com/ChainSafe/gossamer/lib/common"
+	"github.com/ChainSafe/gossamer/pkg/scale"
+)
+
+// WarpProofRequest is a struct for p2p warp proof request
+type WarpProofRequest struct {
+	Begin common.Hash
+}
+
+// Decode decodes the message into a WarpProofRequest
+func (wsr *WarpProofRequest) Decode(in []byte) error {
+	reader := bytes.NewReader(in)
+	sd := scale.NewDecoder(reader)
+	err := sd.Decode(&wsr)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Encode encodes the warp sync request
+func (wsr *WarpProofRequest) Encode() ([]byte, error) {
+	buffer := bytes.NewBuffer(nil)
+	encoder := scale.NewEncoder(buffer)
```

Comment:
For both encoding and decoding here, do we need to create the encoder/decoder object? Can we not just call scale.Marshal and scale.Unmarshal?